### DISCUSSION TITLE
Ikemen Tools Request

### DISCUSSION THREAD

**Discussion by Al-fata-tayeb** [ikemen-engine/Ikemen-GO#1304] [Other]
It's possible to make a Tool can convert mugen 1.1b screenpack to ikemen ? 

It's possible to make a Tool can manage ikemen system ( char , main menu , select menu .... ) without go to .def ? ( like mugen configurator)

It's possible to make a Tool for bitmap font maker (.fnt editor) with his code editor  ?

That it you can choose one of the Tools or just reply about it. I don't force you to do it , I just want to see more information and I wish I see this Tools in the future  .

Thanks , Al-Fata Tayeb ( arabian mugen/ikemen fan )

**Comment by assemblaj** [comment]
Hey so it's funny you asked about this. I took a little detour from IKEMEN to learn Rust, which is sort of known for it's ability to create great Game Tools.  I initially intended to attempt to create a new engine, however seeing the great strides that people have made within IKEMEN in the time that I was gone made me realize that the engine itself is something truly special that needs to be upheld.  So, I figure we can use these tools to sort of support IKEMEN. 

I've been thinking about creating an editor for IKEMEN, not quite an Open Source Fighter Factory but something more modern, that would encompass asset / build management too. I tried to embed IKEMEN directly into a Rust program and it basically just wasn't possible. But I already have the basics of an engine up so I think I can repurpose that as an editor pane and I might be good to go. 

Here's the code for the [IKEMEN-Rust](https://github.com/assemblaj/ik-ggez) engine idea I had, I have [a lot](https://github.com/assemblaj/ik-ggez/tree/main/src/spec) of [utilities ](https://github.com/assemblaj/ik-ggez/blob/main/src/utils/ini.rs)in there for dealing with files.  I also wrote an [SFF Reader](https://github.com/assemblaj/sff-rs) and an [AIR Reader](https://github.com/assemblaj/air-rs) in Rust as most that were out in Rust weren't compatible with all files for one reason or another. 

So yeah, I'll keep this topic updated as things advance.

**Reply by neolao** [reply]
I'm really interested to have one day command line tools to create/update/extract SFF files
